Missing module dependency yup

I was running strapi 4.10.5 and 4.10.6. Previously, things worked as expected, but today, when I had to revert and downgrade strapi to 4.10.5 due to a bug in 4.10.6, I could no longer start up strapi due to a missing yup module dependency.

```
Error: Could not load js config file /Users/asans/proj/vpm-marketing/apps/strapi-cms/node_modules/strapi-plugin-publisher/strapi-server.js: Cannot find module 'yup'
```

The issue stems from the `server/config/schema.js` file requiring yup module, which is not included as a dependency in package.json.

Perhaps there was an implicit assumption that strapi or some other module happened to include this module and it worked.  But that module perhaps took out this dependency, which then started causing this problem.

Regardless of the issue, since this is a dependency for the plugin, it should include it in dependencies.  While hoisting could have covered this problem in npm or yarn, pnpm would have split out errors.

--- a/package.json
+++ b/package.json
@@ -26,20 +26,23 @@
 		"url": "https://github.com/ComfortablyCoding/strapi-plugin-publisher/issues"
 	},
 	"dependencies": {
-		"yarn": "^1.22.19"
+		"lodash": "^4.17.21",
+		"react-query": "^3.39.3",
+		"yup": "^0.32.9"
 	},
 	"devDependencies": {
-		"@babel/core": "^7.21.8",
-		"@babel/eslint-parser": "^7.21.8",
-		"@babel/preset-react": "^7.18.6",
-		"eslint": "^8.39.0",
+		"@babel/core": "^7.22.5",
+		"@babel/eslint-parser": "^7.22.5",
+		"@babel/preset-react": "^7.22.5",
+		"eslint": "^8.43.0",
 		"eslint-config-prettier": "^8.8.0",
 		"eslint-plugin-node": "^11.1.0",
 		"eslint-plugin-react": "^7.32.2",
 		"prettier": "^2.8.8"
 	},
 	"peerDependencies": {
-		"@strapi/strapi": "^4.0.0"
+		"@strapi/strapi": "^4.0.0",
+		"react": "^17.0.0 || ^18.0.0"
 	},
 	"strapi": {
 		"displayName": "Publisher",

